Block/BlockContextManager.php

```diff
@@ -214,4 +212,31 @@ protected function setDefaultExtraCacheKeys(BlockContextInterface $blockContext,
             $blockContext->setSetting('extra_cache_keys', $extraCacheKeys);
         }
     }
+
+    private function resolve(BlockInterface $block, $settings)
+    {
+        $optionsResolver = new OptionsResolver();
+
+        $this->configureSettings($optionsResolver, $block);
+
+        $service = $this->blockService->get($block);
+        $service->setDefaultSettings($optionsResolver, $block);
+
+        $reflector = new \ReflectionMethod($service, 'setDefaultSettings');
```

Comment:
You can store an array with the result of the reflection and still send a trigger. Or at least store the result for one service to avoid doing the same over and over if a Page is composed of the same block.